starter_code/queries.md

```diff
@@ -4,80 +4,93 @@
 
-<!-- Your Code Goes Here -->
+db.companies.find( { "acquisition.price_amount": {$gte: 100000000}, founded_year: {$lte: 2010 } }, { name: 1, ipo:1})
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
-<!-- Your Code Goes Here -->
+db.companies.find( { category_code: null})
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
-<!-- Your Code Goes Here -->
+db.companies.find( { number_of_employees: {$gte: 100, $lt: 1000 }},{ name: 1, number_of_employees:1})
 
 ### 9. Order all the companies by their IPO price in a descending order.
 
-<!-- Your Code Goes Here -->
+db.companies.find().sort({"ipo.valuation_amount": -1})
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+db.companies.find().sort( { number_of_employees: -1 }).limit(10)
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+db.companies.find( { founded_month: {$gt: 6}}.limit(1000)
+
+### 12. All the companies that have been 'deadpooled' after the third year.
 
-<!-- ### 12. All the companies that have been 'deadpooled' after the third year. -->
+db.companies.aggregate({ $project: { '_id': 0, name:1,deadpooled_year:1, difference: {$subtract:["$deadpooled_year", "$founded_year"]}
+}
+    },
+    { $match: { difference: { $lt: 3 }              }
+
+    })
 
-<!-- Your Code Goes Here -->
 
 ### 12. All the companies founded before 2000 that have an acquisition amount of more than 10.000.000
 
-<!-- Your Code Goes Here -->
+db.companies.find( { "acquisition.price_amount": {$gte: 10000000}, founded_year: {$lt: 2000 } })
+
 
 ### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
 
-<!-- Your Code Goes Here -->
+db.companies.find( { "acquisition.acquired_year": {$gt: 2010}, {name:1, acquisition:1}).sort( { "acquisition.price_amount": -1 })
+
 
 ### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
 
-<!-- Your Code Goes Here -->
+db.companies.find({name:1, founded_year:1}).sort( { founded_year: -1 })
 
 ### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` in a descending order. Limit the search to 10 documents.
 
-<!-- Your Code Goes Here -->
+db.companies.find( { founded_day: {$lte: 7}}).sort( { "acquisition.price_amount": -1 }).limit(10)
 
 ### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascending order.
 
-<!-- Your Code Goes Here -->
+db.companies.find( { category_code: "web", number_of_employees: {$gt: 7}}).sort( { number_of_employees: -1 })
```

Comment:
more than 4000 employer ;) 
{$and: [{category_code: "web"}, {number_of_employees: {$gt: 4000}}]}